### PR TITLE
Add Source design system to rule manager client

### PR DESCRIPTION
**Pull request by tjsilver**
## What does this change?
Adds the Source design system and a selection of components, including layout ones, to the rule manager client for testing. 

The components have been added without css overrides, but should be able to be customised to fit in with our other tools if necessary.

NB: This only currently works with Emotion v10 (not v11). React v16 is used as this is required by Source components.

## How to test
- Download branch
- Run `sbt "ruleManager/run"`
- Navigate to: https://typerighter.local.dev-gutools.co.uk/ 

## How can we measure success?
Developers can implement Source design system components in the rule manager client.

## Images

image

### With [Guardian Grid Overlay](https://chrome.google.com/webstore/detail/grid-overlay/nodanebgbfpfpbchadachbkeehidccnl) on

image

### CODE REVIEW

File: rule-manager/rule-manager-client/package.json [hunk 1]
```diff
@@ -12,26 +12,33 @@
...
   },
   "dependencies": {
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1"
+    "@emotion/core": "^10.0.14",
+    "@guardian/src-button": "^2.7.1",
+    "@guardian/src-foundations": "^2.7.1",
+    "@guardian/src-layout": "^2.7.1",
+    "@guardian/src-text-input": "^2.7.1",
+    "emotion-theming": "^11.0.0",
```
Comment (jonathonherbert) [inline rule-manager/rule-manager-client/package.json:39]: Is this used below?

File: rule-manager/rule-manager-client/package.json [hunk 2]
```diff
@@ -12,26 +12,33 @@
...
     "css-loader": "^5.0.1",
     "html-webpack-plugin": "^4.5.0",
     "style-loader": "^2.0.0",
-    "ts-loader": "^8.0.11",
+    "ts-loader": "^8.0.12",
     "tsconfig-paths-webpack-plugin": "^3.3.0",
-    "typescript": "^4.1.2",
-    "webpack": "^5.10.0",
+    "typescript": "^4.1.3",
+    "webpack": "^4.44.2",
```
Comment (jonathonherbert) [inline rule-manager/rule-manager-client/package.json:29]: Presume this downgrade is necessary, but it'd good to document why!

**Comment by tjsilver** [comment]
[Add design system components to rule manager](https://trello.com/c/gjRJODM1/538-add-design-system-components-to-rule-manager)

File: rule-manager/rule-manager-client/src/components/App.tsx [hunk 3]
```diff
@@ -1,7 +1,113 @@
...
+
+const App = () => {
+  const [state, setState] = useState("");
+
+  return (
+    <Fragment>
+      <Container>
+        <Columns>
+          <Column width={4 / 4}>
+            <h1 css={h1}>Typerigher rule manager</h1>
```
Comment (jonathonherbert) [inline rule-manager/rule-manager-client/src/components/App.tsx:35]: Unsure if intentional 😁

**Comment by jonathonherbert** [comment]
@tjsilver a few questions before I approve but this looks great – an Ed Tools first! 👏

File: rule-manager/rule-manager-client/package.json (hunk 2 above)
Comment (tjsilver) [reply rule-manager/rule-manager-client/package.json:29]: It's required for `webpack-dev-middleware`, which is a dep of `webpack-dev-server`, but need to check to see if using v5 breaks anything, will investigate further.

File: rule-manager/rule-manager-client/src/components/App.tsx (hunk 3 above)
Comment (tjsilver) [reply rule-manager/rule-manager-client/src/components/App.tsx:35]: Need to run it through Typerighter :)

File: rule-manager/rule-manager-client/package.json (hunk 1 above)
Comment (tjsilver) [reply rule-manager/rule-manager-client/package.json]: No, have removed it for now.

File: rule-manager/rule-manager-client/package.json (hunk 2 above)
Comment (tjsilver) [reply rule-manager/rule-manager-client/package.json]: I upgraded to webpack 5 again, and removed `webpack-dev-server` for now, as it appears to be unused.

File: rule-manager/rule-manager-client/conf/webpack.common.config.js [hunk 4]
```diff
@@ -52,11 +52,7 @@ module.exports = {
       },
       {
         test: /\.(woff|woff2|eot|ttf|otf)$/,
-        use: [
-          {
-            loader: 'file-loader',
-          },
-        ],
+        type: 'asset/resource'
```
Comment (tjsilver) [inline rule-manager/rule-manager-client/conf/webpack.common.config.js:55]: Asset modules are a Webpack 5 feature

**Review by jonathonherbert — APPROVED** [review]
Great work @tjsilver! 👏

**Pull request merged**